### PR TITLE
feat: Replace NSString for eventId with SentryId

### PR DESCRIPTION
**Pull request by philipphofmann**
## :scroll: Description

Add a new type SentryId that replaces the NSString for eventID. When the sending
of an event fails for example in the SentrySDK, SentryHub or SentryClient
SentryId.empty instead of nil is returned.

## :bulb: Motivation and Context

Add a type for SentryId and avoid returning nil.

## :green_heart: How did you test it?
Unit tests.

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify the changes
- [x] I've updated the CHANGELOG
- [ ] No breaking changes

### CODE REVIEW

File: Sources/Sentry/SentryId.m [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+- (NSUInteger)hash
+{
+    return [self.uuid hash];
+}
+
++ (SentryId *)empty
+{
+    return [[SentryId alloc] initWithUUIDString:emptyUUIDString];
```
[the remark this answers is not in the document]
Comment (philipphofmann) [reply Sources/Sentry/SentryId.m]: Very good point, thanks @bruno-garcia

File: Sources/Sentry/SentryId.m [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    if (nil != uuid) {
+        return [self initWithUUID:uuid];
+    } else {
+        return [self initWithUUIDString:emptyUUIDString];
+    }
+}
+
+- (NSString *)sentryIdString;
+{
+    return self.uuid.UUIDString;
```
[the remark this answers is not in the document]
Comment (philipphofmann) [reply Sources/Sentry/SentryId.m:45]: The [docs of envelopes](https://develop.sentry.dev/sdk/envelopes/#prerequisites.) state a UUID must contain `-`, but below in the examples I see [event_id without `-`](https://develop.sentry.dev/sdk/envelopes/#full-examples). I'm confused now.  I think the backend understands both with or without `-`. When I search for an issue it works for me if I also put in the `-`. I had a discussion with @marandaneto and we both thought it is not necessary anymore.  
I'm going to double-check that.

[the remark this answers is not in the document]
Comment (HazAT) [reply Sources/Sentry/SentryId.m:45]: All event_ids need to be uuid4, without `-` and all lowercase.

[the remark this answers is not in the document]
Comment (philipphofmann) [reply Sources/Sentry/SentryId.m:45]: Just had a discussion with @jan-auer and he just opened up a PR https://github.com/getsentry/develop/pull/110 to clarify that it is recommended to use UUID4 without `-`.

**Pull request merged**